FormidableLabs/urql-devtools
# Update visual regression testing

## About

Cosmos now has synchronous utilities (https://github.com/react-cosmos/react-cosmos/pull/1171)!

## Changes

 - Improve determinism 
   - test up to 5 times when a snapshot fails
   - explicitly set mouse position before snapshot
   - exclude fixtures with "[no snapshot]" in their name
   - run tests in band
   - add delay for loading animations
 - Use `describe.each` to test snapshots (one failing snapshot will no longer prevent all tests from being run)
 - Use fixture file and fixture names for snapshots (fixes vscode not opening images)
 - Use environment variables to specify an alternate endpoint
 - Remove container after visual regression run (`run --rm`)
 - Use `data-snapshot` to snapshot target element + reduce image sizes
